sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FailureSafeParser.scala

```diff
@@ -65,7 +66,8 @@ class FailureSafeParser[IN](
         case DropMalformedMode =>
           Iterator.empty
         case FailFastMode =>
-          throw e.cause
+          throw new SparkException("Malformed records are detected in record parsing. " +
+            s"Parse Mode: ${FailFastMode.name}. Reasons: ${e.getMessage}", e.cause)
```

Comment:
the `Reasons: ${e.getMessage}` is duplicated, the stack trace will include the cause exception.